documentation/development/steps_list.md

```diff
@@ -2,28 +2,55 @@
 
 _The following refers to all commands except `git-new-pull-request`, `git-repo` and `git-town`._
 
+The individual steps that each Git Town command executes are represested via the
+[command pattern](https://en.wikipedia.org/wiki/Command_pattern).
+This allows fully automated and robust `--continue`, `--abort`,
+and `--undo` functionality for each command.
+
+To distinguish the command-pattern commands from the Git Town commands in [src/cmd](../../src/cmd),
+we'll call the former `steps` from now on.
+
+
+## Running commands
+
 Each Git Town command begins by inspecting the current state of the Git repository
 (which branch you are on, whether you have open changes).
 If there are no errors, it generates a list of steps to run.
-Each step is a go `struct`.
 
-When executing a step, the undo steps are determined and added to a separate list.
-This is done by calling the methods `step.CreateUndoStepBeforeRun()` and `step.CreateUndoStepAfterRun()`.
+Steps, located in [src/steps](../../src/steps),
+implement the individual steps that each Git Town command performs,
+like for example [changing to a different Git branch](../../src/steps/checkout_branch_step.go)
+or [pulling down updates for the current branch](../../src/steps/pull_branch_step.go).
+They are Go structs that have a `Run` method which executes the step.
 
-## Abort / Continue
 
-If a Git command fails (typically due to a merge conflict), then the program halts
-and asks the user what they would like to do. In most cases they can either abort or continue.
+## Continuing commands
 
-If the user aborts then `step.CreateAbortStep()` is called and the undo steps are executed.
+If a Git command fails (typically due to a merge conflict),
+then the program halts and asks the user
+whether they would like to abort or continue.
 
-If the user resolves the issue and continues then `step.CreateContinueStep()` is called
-and we resume executing steps.
+If the user resolves the issue and continues,
+Git Town calls the `CreateContinueStep` method of the current (failed) step,
+then continues the remaining step list.
```

Comment:
We create both the continue and abort steps when the conflicts happen. We only run each based on what the user decides to do.

Reply:
done